airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/task/InputConsumerTask.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.task
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
+import io.airbyte.cdk.command.DestinationCatalog
+import io.airbyte.cdk.command.DestinationConfiguration
+import io.airbyte.cdk.command.DestinationStream
+import io.airbyte.cdk.message.CheckpointMessage
+import io.airbyte.cdk.message.CheckpointMessageWrapped
+import io.airbyte.cdk.message.Deserializer
+import io.airbyte.cdk.message.DestinationMessage
+import io.airbyte.cdk.message.DestinationRecord
+import io.airbyte.cdk.message.DestinationRecordWrapped
+import io.airbyte.cdk.message.DestinationStreamAffinedMessage
+import io.airbyte.cdk.message.DestinationStreamComplete
+import io.airbyte.cdk.message.DestinationStreamIncomplete
+import io.airbyte.cdk.message.GlobalCheckpoint
+import io.airbyte.cdk.message.GlobalCheckpointWrapped
+import io.airbyte.cdk.message.MessageQueueSupplier
+import io.airbyte.cdk.message.QueueWriter
+import io.airbyte.cdk.message.StreamCheckpoint
+import io.airbyte.cdk.message.StreamCheckpointWrapped
+import io.airbyte.cdk.message.StreamCompleteWrapped
+import io.airbyte.cdk.message.StreamRecordWrapped
+import io.airbyte.cdk.message.Undefined
+import io.airbyte.cdk.state.MemoryManager
+import io.airbyte.cdk.state.Reserved
+import io.airbyte.cdk.state.SyncManager
+import io.airbyte.cdk.util.use
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Secondary
+import jakarta.inject.Singleton
+import java.io.InputStream
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.FlowCollector
+
+interface InputConsumerTask : SyncTask
+
+/**
+ * Routes @[DestinationStreamAffinedMessage]s by stream to the appropriate channel and @
+ * [CheckpointMessage]s to the state manager.
+ *
+ * TODO: Handle other message types.
+ */
+@SuppressFBWarnings(
```

Comment:
just curious - what's this needed for? I don't actually see any `@Inject` fields, which would have been my first guess

Reply:
I think this one was from when spot bugs couldn't grasp that the deserializer didn't return null? But usually it's because it's confused by the implicit continuation at the end of suspend functions.